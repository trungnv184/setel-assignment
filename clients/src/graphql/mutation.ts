import { gql } from '@apollo/client';

const ADD_PRODUCT_TO_CART = gql`
  mutation AddToCart($cartInput: CartInput) {
    addToCart(cartInput: $cartInput) {
      id
      name
      url
      price
    }
  }
`;

const CREATE_ORDER_PRODUCTS = gql`
  mutation CreateOrder($orderInput: OrderInput) {
    createOrder(orderInput: $orderInput) {
      orderItems {
        id
        name
        url
        price
        quantity
        createdDate
      }
      metadata {
        firstName
        lastName
        address
        phoneNumber
      }
      notes
    }
  }
`;

export { ADD_PRODUCT_TO_CART, CREATE_ORDER_PRODUCTS };
