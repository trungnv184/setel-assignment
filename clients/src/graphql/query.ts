import { gql } from '@apollo/client';

const GET_PRODUCTS = gql`
  query GetProducts {
    products {
      id
      name
      price
      url
      rating
    }
  }
`;

const GET_PRODUCTS_IN_CART = gql`
  query GetProductsCart {
    getProductsCart {
      id
      name
      price
      url
      quantity
    }
  }
`;

const GET_ORDER_BY_ID = gql`
  query GetOrder($orderId: String) {
    getOrder(orderId: $orderId) {
      id
      orderItems {
        id
        name
        price
        url
        quantity
      }
      metadata {
        firstName
        lastName
        address
        phoneNumber
      }
      notes
      state
      createdDate
      updatedDate
    }
  }
`;

const GET_ORDER_STATUS = gql`
  query GetOrderStatus($orderId: String) {
    getOrder(orderId: $orderId) {
      state
    }
  }
`;

export { GET_PRODUCTS, GET_PRODUCTS_IN_CART, GET_ORDER_BY_ID, GET_ORDER_STATUS };
