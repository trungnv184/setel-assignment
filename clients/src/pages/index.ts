export { default as Orders } from './Orders/Orders';
export { default as Products } from './Products/Products';
export { default as Carts } from './Carts/Carts';
export { default as OrderDetail } from './OrderDetail/OrderDetail';
