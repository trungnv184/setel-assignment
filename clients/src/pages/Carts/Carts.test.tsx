import { render, screen, waitFor } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { Carts } from 'pages';
import { GET_PRODUCTS_IN_CART } from 'graphql/query';
import { GET_PRODUCTS_IN_CART_MOCK } from 'test/mocks/carts.mock';

let isCalled = false;
const mocks = [
  {
    request: {
      query: GET_PRODUCTS_IN_CART,
    },
    result: () => {
      isCalled = true;
      return GET_PRODUCTS_IN_CART_MOCK;
    },
  },
];
describe('CartsPage', () => {
  test('Should render carts page', async () => {
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Carts />
      </MockedProvider>
    );

    await waitFor(() => {
      expect(isCalled).toBeTruthy();
    });

    expect(screen.getByText(/Tâm lý học hạnh phúc/i)).toBeInTheDocument();
    expect(screen.getByText(/Dạy con học ít hiểu nhiều/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
