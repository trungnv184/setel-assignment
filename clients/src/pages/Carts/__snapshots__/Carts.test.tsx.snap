// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartsPage Should render carts page 1`] = `
<div>
  <div
    class="ui secondary segment"
  >
    <form
      class="ui form"
    >
      <div
        class="ui error message"
      >
        <div
          class="content"
        >
          <div
            class="header"
          >
            Form Validation
          </div>
        </div>
      </div>
      <div
        class="equal width fields"
      >
        <div
          class="field"
        >
          <label>
            First name
          </label>
          <div
            class="ui input"
          >
            <input
              name="firstName"
              placeholder="First name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="field"
        >
          <label>
            Last name
          </label>
          <div
            class="ui input"
          >
            <input
              name="lastName"
              placeholder="Last name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="field"
        >
          <label>
            Phone Number
          </label>
          <div
            class="ui input"
          >
            <input
              name="phoneNumber"
              placeholder="Enter Phone Number"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <label>
          Address
        </label>
        <div
          class="ui input"
        >
          <input
            name="address"
            placeholder="Enter your address to deliver"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label>
          Notes
        </label>
        <textarea
          name="notes"
          placeholder="Add your special notes"
          rows="3"
        />
      </div>
      <div
        class="field"
      >
        <button
          class="ui blue button"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
  <table
    class="ui celled padded table"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class="single line"
        >
          Image
        </th>
        <th
          class=""
        >
          Name
        </th>
        <th
          class=""
        >
          Price
        </th>
        <th
          class=""
        >
          Quantity
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
      >
        <td
          class=""
        >
          <img
            class="ui small image"
            src="https://salt.tikicdn.com/cache/280x280/ts/product/5c/07/92/7ac1298303440baa2ed74fa20c9f7385.jpg"
          />
        </td>
        <td
          class="single line"
        >
          Làm chủ cảm xúc
        </td>
        <td
          class=""
        >
          200.000 ₫
        </td>
        <td
          class=""
        >
          1
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class=""
        >
          <img
            class="ui small image"
            src="https://salt.tikicdn.com/cache/280x280/ts/product/59/e3/e7/642ce38017ac1d35f2e87234b6fd56f2.jpg"
          />
        </td>
        <td
          class="single line"
        >
          Tâm lý học hạnh phúc
        </td>
        <td
          class=""
        >
          200.000 ₫
        </td>
        <td
          class=""
        >
          1
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class=""
        >
          <img
            class="ui small image"
            src="https://salt.tikicdn.com/cache/280x280/ts/product/5f/36/dd/c64c91016b320ae4591efcf93d0ebf30.jpg"
          />
        </td>
        <td
          class="single line"
        >
          Dạy con học ít hiểu nhiều
        </td>
        <td
          class=""
        >
          200.000 ₫
        </td>
        <td
          class=""
        >
          1
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class=""
        >
          <img
            class="ui small image"
            src="https://salt.tikicdn.com/cache/280x280/ts/product/95/c5/71/f260011660001af960c42d7d2f1c5dc1.jpg"
          />
        </td>
        <td
          class="single line"
        >
          Đừng chạy theo số đông
        </td>
        <td
          class=""
        >
          200.000 ₫
        </td>
        <td
          class=""
        >
          1
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
